internal/states/statefile/read_test.go

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: MPL-2.0
+
+package statefile
+
+import (
+	"errors"
+	"os"
+	"testing"
+)
+
+func TestReadErrNoState(t *testing.T) {
+	emptyFile, err := os.Open("testdata/read/empty")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer emptyFile.Close()
+
+	_, err = Read(emptyFile)
+	if !errors.Is(err, ErrNoState) {
+		t.Fatalf("expected ErrNoState, got %T", err)
+	}
+
+	nilFile, err := os.Open("")
+	if err == nil {
+		t.Fatal("wrongly succeeded in opening non-existent file")
+	}
+
```

Comment:
do we really need that check? as `os` is golang internal package

Reply:
I didn't put it there to test `os`, I put it there to communicate to future devs what the test is up to. I could change this to a comment, though having it like this puts up a speed-bump to having the test accidentally neutered.